File: cpp/example_code/codebuild/build_project.cpp
// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
// SPDX-License-Identifier: Apache-2.0

#include <aws/core/Aws.h>
#include <aws/core/utils/Outcome.h>
#include <aws/codebuild/CodeBuildClient.h>
#include <aws/codebuild/model/StartBuildRequest.h>
#include <aws/codebuild/model/StartBuildResult.h>
#include <iostream>

/**
 * Starts the project build based on command line input
 */

int main(int argc, char **argv)
{
  if (argc != 2)
  {
    std::cout << "Usage: build_project <project_name>";
    return 1;
  }
  Aws::SDKOptions options;
  Aws::InitAPI(options);
  {
    Aws::String project_name(argv[1]);
    Aws::CodeBuild::CodeBuildClient codebuild;

    Aws::CodeBuild::Model::StartBuildRequest sb_req;
    sb_req.SetProjectName(project_name);

    auto sb_out = codebuild.StartBuild(sb_req);

    if (sb_out.IsSuccess())
    {
      std::cout << "Successfully started build" << std::endl;
    }

    else
    {
      std::cout << "Error starting build" << sb_out.GetError().GetMessage()
        << std::endl;
    }
  }

  Aws::ShutdownAPI(options);
  return 0;
}

File: ruby/example_code/codebuild/build-project.rb
# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
# SPDX-License-Identifier: Apache-2.0

# snippet-start:[codebuild.Ruby.buildProject]
require "aws-sdk-codebuild"  # v2: require 'aws-sdk'

project_name = ""

if ARGV.length != 1
  puts "You must supply the name of the project to build"
  exit 1
else
  project_name = ARGV[0]
end

client = Aws::CodeBuild::Client.new

begin
  client.start_build(project_name: project_name)
  puts "Building project " + project_name
rescue StandardError => ex
  puts "Error building project: " + ex.message
end
# snippet-end:[codebuild.Ruby.buildProject]
